lib/node_modules/@stdlib/math/base/special/gamma-delta-ratio/src/main.c

```diff
@@ -0,0 +1,172 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C++ code and copyright notice are from the [Boost library]{@link http://www.boost.org/doc/libs/1_85_0/boost/math/special_functions/gamma.hpp}. The implementation has been modified for JavaScript.
+*
+* ```text
+* Copyright John Maddock 2006-7, 2013-14.
+* Copyright Paul A. Bristow 2007, 2013-14.
+* Copyright Nikhar Agrawal 2013-14.
+* Copyright Christopher Kormanyos 2013-14.
+*
+* Use, modification and distribution are subject to the
+* Boost Software License, Version 1.0. (See accompanying file
+* LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+* ```
+*/
+
+#include "stdlib/math/base/special/gamma_delta_ratio.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/math/base/special/floor.h"
+#include "stdlib/math/base/special/gamma.h"
+#include "stdlib/math/base/special/factorial.h"
+#include "stdlib/constants/float64/max_safe_nth_factorial.h"
+#include "stdlib/math/base/special/gamma_lanczos_sum.h"
+#include "stdlib/math/base/special/log1p.h"
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/constants/float64/eps.h"
+#include "stdlib/constants/float64/e.h"
+#include "stdlib/constants/float64/gamma_lanczos_g.h"
+
+static const double FACTORIAL_169 = 4.269068009004705e+304;
+
+/**
+* Calculates the ratio of two gamma functions via Lanczos approximation.
+*
+* ## Notes
+*
+* -   When \\( z < \epsilon \\), we get spurious numeric overflow unless we're very careful. This can occur either inside `lanczosSum(z)` or in the final combination of terms. To avoid this, split the product up into 2 (or 3) parts:
+*
+*     ```tex
+*     \begin{align*}
+*     G(z) / G(L) &= 1 / (z \cdot G(L)) ; z < \eps, L = z + \delta = \delta \\
+*     z * G(L) &= z * G(lim) \cdot (G(L)/G(lim)) ; lim = \text{largest factorial}
+*     \end{align*}
+*     ```
+*
+* @private
+* @param z        first gamma parameter
+* @param delta    difference
+* @returns        gamma ratio
+*/
+static double gammaDeltaRatioLanczos( const double z, const double delta ) {
+	double result;
+	double ratio;
+	double zgh;
+
+	if ( z < STDLIB_CONSTANT_FLOAT64_EPS ) {
+		if ( delta > STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FACTORIAL ) {
```

Comment:
```suggestion
		if ( delta >= STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FACTORIAL ) {
```

This should be `>=` as the Boost implementation does `boost::math::max_factorial<T>::value < delta`, which means that the branch should be exercised when `delta` is either equal to or greater than. If this is the same in the JS implementation, that should be fixed there, as well.